modules/st2-elastic-field/directive.js

```diff
@@ -0,0 +1,177 @@
+          dimensions.outer = {
+            width:  window.parseInt(computedStyle.getPropertyValue('padding-left'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('padding-right'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-left-width'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-right-width'), 10),
+            height: window.parseInt(computedStyle.getPropertyValue('padding-top'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('padding-bottom'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-top-width'), 10) +
+                    window.parseInt(computedStyle.getPropertyValue('border-bottom-width'), 10)
+          };
+        } else {
+          dimensions.outer = {
+            width: 0,
+            height: 0
+          };
+        }
+            
+        dimensions.maxHeight = dimensions.maxHeight > 0 ? dimensions.maxHeight : Infinity;
+        
+        if ($textarea.minRows) {
+          dimensions.minHeight = window.Math.max(dimensions.minHeight,
+                                                 $textarea.minRows * lineHeight + dimensions.outer.height);
+        }
+        
+        if ($textarea.maxRows) {
+          dimensions.maxHeight = window.Math.min(dimensions.maxHeight,
+                                                 $textarea.maxRows * lineHeight + dimensions.outer.height);
+        }
+        
+        return dimensions;
+      }
+      
+      function imitate(source, target) {
+        var computedStyle = window.getComputedStyle(source),
+            propertiesToCopy = [
+              'box-sizing',
+              'width',
+              'padding-top',
+              'padding-right',
+              'padding-bottom',
+              'padding-left',
+              'border-top-width',
+              'border-right-width',
+              'border-bottom-width',
+              'border-left-width',
+              'font-family',
+              'font-size',
+              'font-weight',
+              'font-style',
+              'letter-spacing',
+              'line-height',
+              'text-transform',
+              'word-spacing',
+              'text-indent'
+            ],
+            styleString = '';
+         
+        angular.forEach(propertiesToCopy, function(property) {
+          styleString += property + ':' + computedStyle.getPropertyValue(property) + ';';
+        });
+        
+        target.setAttribute('style', styleString);
+        target.value = source.value;
+      }
+      
+      function adjust($textarea, $clone, force) {
+        if ($textarea.isBeingAdjusted && !force) {
+          return;
+        }
+        
+        var dimensions = getDimensions($textarea),
+            overflow = 'hidden',
+            cloneHeight;
+        
+        $textarea.isBeingAdjusted = true;
```

Comment:
I guess you are right. The flag is there from the library the code is based on. I wanted to review the flag after I rewrite the code, but forgot about it.